engine/classes/ElggWidgetDefinition.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/**
+ * \ElggWidgetDefinition
+ *
+ * Helper class for defining a widget
+ *
+ * @package    Elgg.Core
+ * @subpackage Widgets
+ *
+ */
+class ElggWidgetDefinition {
+
+	/**
+	 * @var string Handler of the widget
+	 */
+	public $handler;
+	
+	/**
+	 * \ElggWidgetDefinition constructor
+	 *
+	 * @param string $handler Handler of the widget
```

Comment:
because the ElggWidgetDefinition objects will be returned instead of the current stdClass objects when you call elgg_get_widget_types... People will expect ->handler to still work on these objects